### PR TITLE
mysql,terror: add plugin error msg

### PR DESCRIPTION
**Pull request by lysu**
add plugin error message for https://github.com/pingcap/tidb/pull/8788

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/parser/173)

### CODE REVIEW

**Comment by tiancaiamao** [comment]
Those errors is defined for TiDB only, and it's better to move it back to TiDB repo.
If Go2.0 error come out, I'll take the refactor into consideration...

File: mysql/errname.go [hunk 1]
```diff
@@ -916,6 +916,16 @@ var MySQLErrName = map[uint16]string{
 	ErrForUpdateCantRetry: "[%d] can not retry select for update statement",
 	ErrAdminCheckTable:    "TiDB admin check table failed.",
 
+	ErrInvalidPluginID:            "Wrong plugin id: %s, valid plugin id is [name]-[version], both name and version should not contain '-'",
+	ErrInvalidPluginManifest:      "Can not read plugin %s's manifest",
```
Comment (kennytm) [inline mysql/errname.go]:
```suggestion
	ErrInvalidPluginManifest:      "Cannot read plugin %s's manifest",
```

File: mysql/errname.go [hunk 2]
```diff
@@ -916,6 +916,16 @@ var MySQLErrName = map[uint16]string{
...
+	ErrInvalidPluginName:          "Plugin load with %s but got wrong name %s",
+	ErrInvalidPluginVersion:       "Plugin load with %s but got %s",
+	ErrDuplicatePlugin:            "Plugin [%s] be duplicate declared in init parameters",
+	ErrInvalidPluginSysVarName:    "Plugin %s's sysVar %s must be prefix with plugin name %s",
```
Comment (kennytm) [inline mysql/errname.go]:
```suggestion
	ErrInvalidPluginSysVarName:    "Plugin %s's sysVar %s must start with its plugin name %s",
```

File: mysql/errname.go [hunk 3]
```diff
@@ -916,6 +916,16 @@ var MySQLErrName = map[uint16]string{
 	ErrForUpdateCantRetry: "[%d] can not retry select for update statement",
 	ErrAdminCheckTable:    "TiDB admin check table failed.",
 
+	ErrInvalidPluginID:            "Wrong plugin id: %s, valid plugin id is [name]-[version], both name and version should not contain '-'",
+	ErrInvalidPluginManifest:      "Can not read plugin %s's manifest",
+	ErrInvalidPluginName:          "Plugin load with %s but got wrong name %s",
+	ErrInvalidPluginVersion:       "Plugin load with %s but got %s",
+	ErrDuplicatePlugin:            "Plugin [%s] be duplicate declared in init parameters",
```
Comment (kennytm) [inline mysql/errname.go]:
```suggestion
	ErrDuplicatePlugin:            "Plugin [%s] is redeclared",
```

I'm not sure how "declared in init parameters" is relevant besides this error is emitted in the `plugin.Init()` function, so I've removed the last clause in the suggestion.

File: mysql/errname.go [hunk 4]
```diff
@@ -916,6 +916,16 @@ var MySQLErrName = map[uint16]string{
...
+	ErrInvalidPluginSysVarName:    "Plugin %s's sysVar %s must be prefix with plugin name %s",
+	ErrRequireVersionCheckFail:    "Plugin %s require %s be %v but got %v",
+	ErrUnsupportedReloadPlugin:    "Plugin %s isn't loaded so can not reload",
```
Comment (kennytm) [inline mysql/errname.go]:
```suggestion
	ErrUnsupportedReloadPlugin:    "Plugin %s isn't loaded so cannot be reloaded",
```

**Pull request merged**